apps/opik-backend/src/main/java/com/comet/opik/domain/ExperimentDAO.java

```diff
@@ -400,14 +400,23 @@ AND ilike(name, CONCAT('%', :name, '%'))
 
     private static final String FIND_EXPERIMENT_DATASET_ID_EXPERIMENT_IDS = """
             SELECT
-                dataset_id
+                distinct dataset_id
             FROM experiments
             WHERE id IN :experiment_ids
             AND workspace_id = :workspace_id
             ORDER BY id DESC, last_updated_at DESC
             LIMIT 1 BY id
             ;
             """;
+    public static final String GET_ALL_DATASET_IDS_FROM_EXPERIMENTS = """
```

Comment:
This new query is just a copy and paste of the one above, but without the filter on IDs. Just make that part optional in the where with a template.